python/pyspark/taskcontext.py

```diff
@@ -88,3 +89,9 @@ def taskAttemptId(self):
         TaskAttemptID.
         """
         return self._taskAttemptId
+
+    def getLocalProperty(self, key):
+        """
+        Get a local property set upstream in the driver, or None if it is missing.
```

Comment:
If it's missing it will result in a `KeyError`, maybe you want `return self._localProperties.get(key)` which returns `None` as the default?  That seems better to me too, although you might want to add an optional `default` value.

Reply:
Right.

Reply:
Thanks @BryanCutler for catching this stupid stuff. not at comfortable with python.

Reply:
Just get(key) returns `None` if missing tho ..

Reply:
get(key) would return `None` tho if it's missing.

Reply:
yeah. I added the default=None to make it super obvious. i see no harm in making the code more intuitive.

Reply:
Wait .. it's a dictionary. Does a dictionary's `get` has `default` keyword argument?

Reply:
No, it seems not. Shall we change it to `get(key)`?

Reply:
The Java / Scala equivalents of this API return `null` for missing keys, so on the one hand returning `None` is kinda consistent with that.

On the other hand, consider a case where you want to specify an alternative in case a key is not set:

With this API, you might think of doing something like `tc.getLocalProperty('key') or 'defaultValue'`, which potentially could be a problem in case a non-None key could have a `False`-y value. I suppose we're only dealing with strings here, though, and that'd only happen for empty strings. If we allowed non-strings to be returned here, though, then we'd have problems if we're returning values like `0`. For that case, having a `getLocalProperty('key', 'defaultValue')` is a bit more useful.

Reply:
I am trying to match the Java TaskContext API, which only has a `TaskContext.getLocalProperty(key)`

Reply:
FWIW we can always evolve what you have here towards the `getLocalProperty(key, default=None)` case; that evolution would maintain behavior and source compatibility. Therefore maybe it's fine to defer that until later if we're not sure whether that variant is useful.

Reply:
Makes sense but +1 for leaving it out since I either don't know how commonly it will be used for now.

Reply:
Sounds good to me since this matches the Scala API